aiomisc_dependency/__init__.py

```diff
@@ -13,9 +17,38 @@ def dependency(func):
     return STORE.provider(scope='session')(func)
 
 
-async def inject(target, dependencies):
+def _aggregate(*, dependencies_list, dependencies_map) -> dict:
+    dependencies = {name: name for name in dependencies_list}
+
+    overlapping = set(dependencies.keys()) & set(dependencies_map.keys())
+    if overlapping:
+        log.warning(
+            '__dependencies__ and __dependencies_map__ overlap with %s',
+            overlapping
+        )
+
+    dependencies.update(dependencies_map)
+    return dependencies
+
+
+async def inject(target, *, dependencies_list, dependencies_map):
+    dependencies_list = dependencies_list or []
+    dependencies_map = dependencies_map or {}
+    if (
+        isinstance(dependencies_list, Mapping) or
+        not isinstance(dependencies_map, Mapping)
+    ):
+        raise ValueError(
+            '__dependencies__ must be not be mapping, '
```

Comment:
Typo: `must not be mapping`.
Better message: `__dependencies__ must be Sequence, not Mapping`

Reply:
Not sure, `set` is not a sequence, for example

Reply:
Ok, let's it be `Iterable`

Reply:
Not a good practice, for example, generators are iterables

Reply:
Forced to be a set or a sequence